src/QueueBatch/Impl/BindingProvider.cs

```diff
@@ -47,9 +57,19 @@ public async Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext
             return new TriggerBinding(context.Parameter, queue, TimeSpan.FromSeconds(attr.MaxBackOffInSeconds), attr.ParallelGets, attr.RunWithEmptyBatch, loggerFactory);
         }
 
-        CloudQueue CreatePoisonQueue(string name)
+        CloudQueue CreatePoisonQueue(CloudQueueClient queueClient, string name)
+        {
+            return queueClient.GetQueueReference(name + PoisonQueueSuffix);
+        }
+
+        private string Resolve(string queueName)
         {
-            return provider.GetClient().GetQueueReference(name + PoisonQueueSuffix);
+            if (nameResolver == null)
+            {
+                return queueName;
+            }
+
+            return nameResolver.ResolveWholeString(queueName);
```

Comment:
As far as I see, this adds a new functionality of using custom expressions for queue names. Is it required for handling different accounts, or maybe it's provided to mimic `Storage` behavior? If the second, I'd prefer to keep it out of this PR.

Reply:
Yes, I see the value of having it resolved, not hard-coded. Of course I'd be happy to accept it as a separate PR.